pipelines/algolia_hn_search/__init__.py

```diff
@@ -0,0 +1,127 @@
+"""
+	Loads all the search hits for specific keywords from the Algolia Hacker News Search API
+	https://hn.algolia.com/api
+"""
+import dlt
+import time
+
+from datetime import datetime
+from dlt.sources.helpers import requests
+
+
+url = "https://hn.algolia.com/api/v1/search_by_date"
+
+@dlt.source
+def algolia_hn_search(
+	keywords, # list of keywords for which the data needs to be loaded
+	start_date, # start date in datetime or "yyyy-mm-dd" format
+	end_date=datetime.today(), # end date in datetime or "yyyy-mm-dd" format
+	text="any", # possible values: "story","comment". For any other value, everything is loaded.
+	daily_load=False # loads data on a daily schedule when set to True
+):
+
+	"""
+		Source method for the Algolia Hacker News Search API: https://hn.algolia.com/api
+	"""	
+
+	# Read start date as string or datetime and convert it to UNIX timestamp
+	if isinstance(start_date,str):
+		start_timestamp = int(time.mktime(
+			datetime.strptime(start_date,"%Y-%m-%d").timetuple()
+		))
+	else:
+		start_timestamp = int(
+			time.mktime(start_date.timetuple())
+		)
+
+	# Read end date as string or datetime and convert it to UNIX timestamp
+	if isinstance(end_date,str):
+		end_timestamp = int(time.mktime(
+			datetime.strptime(end_date,"%Y-%m-%d").timetuple()
+		))
+	else:
+		end_timestamp = int(
+			time.mktime(end_date.timetuple())
+		)
+
+	today = int(time.mktime(datetime.today().timetuple()))
+
+	# Don't load the data for dates after the current date
+	end_timestamp = min(today,end_timestamp)
+
+	# Ensure that the input start date is smaller than the input end date
+	if start_timestamp > end_timestamp:
+		raise Exception(f"""
+			start_date {start_date} is larger than end_date {end_date}.\n
+			Ensure that the start_date is smaller or equal to than the end_date.\n
+			Also ensure that the start_date is smaller than or equal to today {today}.
+		""")
+
+	# Specify text = "comment" or text="story" when calling the function
+	# to load only comments or stories
+	if text in ["comment","story"]:
+		tags = text
+	# Pass any other value to load everything (default behaviour)
+	else: 
+		tags = '(story,comment)'
+
+	return keyword_hits(keywords,start_timestamp,end_timestamp,tags,daily_load)
+
+
+@dlt.resource(write_disposition="append")
+def keyword_hits(
+	keywords, # list of keywords for which the data needs to be loaded
+	start_timestamp, # UNIX timestamp for the start date
+	end_timestamp, # UNIX timestamp for the end date
+	tags, # parameter for the API call to specify "story", "comment" or "(story,comment)"
+	daily_load=False # loads data on a daily schedule when set to True
```

Comment:
if you want to convert this function to incremental loading see this example:
1. see this example how incremental is added https://github.com/dlt-hub/pipelines/blob/master/pipelines/shopify_dlt/__init__.py#L36
2. define your incremental column on `end_date`. (which you must add to your data)
3. consider converting this function to inner function ie. https://github.com/dlt-hub/pipelines/blob/master/pipelines/shopify_dlt/__init__.py#L36 is an example. then you do not even need to pass the arguments - your function sees the arguments in the source

btw. gpt-4 will do most of the fixes for you IMO